hw6/Names.java

```diff
@@ -0,0 +1,109 @@
+import java.io.*;
+import java.util.*;
+import java.awt.*;
+
+public class Names {
+   public static final int numOfDecades   = 14;
+   public static final int startingYear   = 1880;
+   public static final int widthPerDecade = 70;
+   
+   public static void main(String[] args) throws FileNotFoundException {
+      Scanner input = new Scanner(new File("names.txt"));
+      Scanner console = new Scanner(System.in);
+      giveIntro();
+      //form searchCombo with name+space+sex (uppercase), read from user output.
+      System.out.print("name? ");
+      String name1 = console.nextLine();
+      String name = (name1.substring(0, 1).toUpperCase() + name1.substring(1));
+      System.out.print("sex (M or F)? ");
+      String sex = console.nextLine().toUpperCase();
+      String searchCombo = name + " " + sex;
+      //find searchCombo in input file. If there is a matching output, construct a drawing panel.
+      //If there is no matching output, print not found message.
+      String line = find(input, searchCombo);
+      if (line.length() > 0) {
+         drawGraph (line,searchCombo);
+      } else {
+         System.out.println("name/sex combination not found");
+      }
+   }   
+   // introduces the purpose of the program to the user
+   public static void giveIntro() {
+      System.out.println("This program allows you to search through the");
+      System.out.println("data from the Social Security Administration");
+      System.out.println("to see how popular a particular name has been");
+      System.out.println("since " + startingYear + " .");
+      System.out.println();
+   }   
+   // searches for formatted data that starts with searchCombo
+   // return line if found; return empty if not found
+   public static String find(Scanner input, String searchCombo) {
+      while (input.hasNextLine()) {
+         String line = input.nextLine();
+         //format token, and compare token based with search combo
+         String dataCombo = tokenFormat(line);
+         if (dataCombo.startsWith(searchCombo)) {
```

Comment:
-0.25

1. If you are going down this route, constructing your own combo, you could just use `equals`.
2. Like what I said earlier, if you pass the `name` and `sex` as 2 parameters, you could extract those 2 value from the line, and compare them individually.

but kudos to being creative